### PR TITLE
Master initialize

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
last commit is pulled out of #336.

**Comment by mulkieran** [comment]
@agrover @trgill review?

File: src/engine/strat_engine/pool.rs [hunk 1]
```diff
@@ -225,11 +225,12 @@ impl Pool for StratPool {
             }
         }
 
+        // TODO: Roll back on filesystem initialization failure.
+        let dm = try!(DM::new());
         let pool_uuid = &self.pool_uuid;
```
Comment (trgill) [inline src/engine/strat_engine/pool.rs]: optional request: pool_uuid isn't needed.

Comment (mulkieran) [inline src/engine/strat_engine/pool.rs]: Might as well...

**Comment by mulkieran** [comment]
Changes made. @agrover review?

**Comment by agrover** [comment]
👍 . last reviewer so merging.

**Pull request merged**